### PR TITLE
Tms/address first 8 rubocop offenses

### PR DESCRIPTION
**Pull request by tmikeschu**
## Overview
This continues efforts to clean up the codebase per #85. In the interest of not creating a huge PR, I handled the first 8 offense categories and the associated files, and will continue to make similarly bite-sized PRs. One bite at a time!

All tests pass.

## Details
Offenses handled include: alphabetizing the Gemfile, removing extra spaces, changing deprecated method calls, adding underscores to unused block/method arguments, and deleting unused variable definitions.

### CODE REVIEW

**Comment by tmikeschu** [comment]
Does `.to_s` block a SQL injection? It looks like to of the changes I made are now vulnerable.

**Comment by ericfreese** [comment]
> Does .to_s block a SQL injection? It looks like to of the changes I made are now vulnerable.

Nope. I think those most have been vulnerable before as well but for some reason code climate is just now picking them up.

**Comment by tmikeschu** [comment]
@ericfreese should that fix be managed in this PR, or a different one?

**Comment by ericfreese** [comment]
@tmikeschu I think it should probably be a separate PR

**Comment by ericfreese** [comment]
Would it be possible to remove the merge commit cc3a01f so that the only commit on your branch is the one doing the rubocop cleanup?

Instead of merging `upstream/master` into your branch, you should be able to rebase your branch on `upstream/master`.

Something like:

```shell
$ git fetch upstream
$ git checkout upstream/master
$ git rebase
$ git checkout tms/address-first-8-rubocop-offenses
$ git rebase upstream/master
```

**Comment by tmikeschu** [comment]
@rylanb rebased and reduced to one commit.

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -1,11 +1,16 @@
 # This configuration was generated by
 # `rubocop --auto-gen-config`
+<<<<<<< HEAD
 # on 2017-03-22 20:00:21 -0600 using RuboCop version 0.47.1.
+=======
+# on 2017-03-18 21:04:58 -0600 using RuboCop version 0.47.1.
+>>>>>>> Address first 8 offenses in rubocop_todo.yml
```
Comment (ericfreese) [inline .rubocop_todo.yml]: @tmikeschu Looks like some merge conflict artifacts left over

File: app/controllers/logs_controller.rb [hunk 2]
```diff
@@ -393,38 +393,38 @@ def export
...
+        lp.count = lpdata['count']
+        lp.description = lpdata['description']
+        lp.food_type_id = lpdata['food_type_id'].to_i
+        lp.log_id = log.id
+        lp.weight = lpdata['weight'].to_f
+        ret.push lp
+        lp.save
+      } unless params['log_parts'].nil?
+      ret
+    end
```
Comment (ericfreese) [inline app/controllers/logs_controller.rb]: I don't _think_ we want to indent these methods. I'm getting a rubocop error when running against this branch:


```
app/controllers/logs_controller.rb:396:5: C: Inconsistent indentation detected.
    def parse_and_create_log_parts(params, log) ...
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

May be a result of merging/rebasing?

File: .rubocop_todo.yml (hunk 1 above)
Comment (tmikeschu) [inline .rubocop_todo.yml]: @ericfreese apologies! Not sure why these didn't flag on my GUI.

File: app/controllers/logs_controller.rb (hunk 2 above)
Comment (tmikeschu) [inline app/controllers/logs_controller.rb]: @ericfreese thanks for spotting that. Two of the three were indented originally, and I thought that was the private method convention. Fixed now!

File: Gemfile [hunk 3]
```diff
@@ -2,48 +2,48 @@ source 'https://rubygems.org'
...
-gem 'thin'
 gem 'pg'
+gem 'rails', '~> 3.2.16'
 gem 'rails_12factor'
+gem 'thin'
 
 # for handling json objects with ruby
 gem 'json'
 
 # lets us use the ubiquitous jquery javascript library
```
Comment (cllns) [inline Gemfile]: can we get rid of this line too? It's now misplaced but it doesn't really offer any info. It's pretty clear what `jquery-rails` does :)

Comment (tmikeschu) [inline Gemfile]: @cllns for sure! Thanks for reading this over.

File: app/controllers/logs_controller.rb [hunk 4]
```diff
@@ -44,7 +44,7 @@ def tardy
   end
 
   def index(logs=nil, header='Entire Log')
-    filter = filter.nil? ? '' : " AND #{filter}"
+    filter.nil? ? '' : " AND #{filter}"
```
Comment (rylanb) [inline app/controllers/logs_controller.rb]: Hmmm does this do something I don't understand? Seems like the assignment was lost?

File: app/controllers/logs_controller.rb [hunk 5]
```diff
@@ -84,7 +84,7 @@ def stats
       @transport_per_year[l.name] = [] if @transport_per_year[l.name].nil?
     }
     @transport_per_year.keys.each{ |k|
-      @transport_per_year[k] = @transport_years.collect{ |y| 0 }
+      @transport_per_year[k] = @transport_years.collect{ |_y| 0 }
```
Comment (rylanb) [inline app/controllers/logs_controller.rb]: Could you link me to the _ convention reco from Rubocop?

File: app/models/location.rb [hunk 6]
```diff
@@ -168,35 +168,35 @@ def populate_detailed_hours_json_before_save
...
+      @day_info = {}
+      (0..6).each do |index|
+        prefix = 'day'+index.to_s+'_'
+        next if detailed_hours[index.to_s].nil?
+        start = detailed_hours[index.to_s]['start']
+        stop = detailed_hours[index.to_s]['end']
+        next if start.nil? or stop.nil?
+        write_day_info( prefix+'status', detailed_hours[index.to_s]['status'].to_i )
+        # carefully set start time
+        t = Time.find_zone(self.time_zone).parse( start )
```
Comment (rylanb) [inline app/models/location.rb]: t could be 'time' or something more than 't' but not a huge deal

File: app/controllers/logs_controller.rb (hunk 4 above)
Comment (tmikeschu) [inline app/controllers/logs_controller.rb]: @rylanb I was also confused, it was flagged for not being used, so I deleted the assignment. It seems like if it's not used, this line could be deleted, but I wasn't sure.

File: app/controllers/logs_controller.rb (hunk 5 above)
Comment (tmikeschu) [inline app/controllers/logs_controller.rb]: Sure! [here](http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/UnusedBlockArgument) and [here](http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/UnusedMethodArgument).

File: app/models/location.rb (hunk 6 above)
Comment (tmikeschu) [inline app/models/location.rb]: @rylanb Nice catch! I'll pay closer attention to low hanging fruit like this next time.

File: app/controllers/logs_controller.rb (hunk 4 above)
Comment (ericfreese) [inline app/controllers/logs_controller.rb]: Yeah, probably just want to remove this line entirely.

Comment (tmikeschu) [inline app/controllers/logs_controller.rb]: Done!

**Comment by ericfreese** [comment]
@tmikeschu I'm still seeing some rubocop errors on this branch:

```
$ bundle exec rubocop
Inspecting 186 files
......W.......................C........................................................................................................................C..................................

Offenses:

app/controllers/cell_carriers_controller.rb:11:26: W: Unused method argument - record. If it's necessary, use _ or _record as an argument name to indicate that it won't be used. You can also write as update_authorized?(*) if you want the method to accept any arguments but don't care about them.
  def update_authorized?(record=nil)
                         ^^^^^^
app/controllers/cell_carriers_controller.rb:15:26: W: Unused method argument - record. If it's necessary, use _ or _record as an argument name to indicate that it won't be used. You can also write as delete_authorized?(*) if you want the method to accept any arguments but don't care about them.
  def delete_authorized?(record=nil)
                         ^^^^^^
app/models/location.rb:171:5: C: Inconsistent indentation detected.
    def init_detailed_hours ...
    ^^^^^^^^^^^^^^^^^^^^^^^
app/models/location.rb:190:43: C: Space missing after comma.
        time = time.change(:year=>now.year,:month=>now.month, :day=>now.day)
                                          ^
app/models/location.rb:194:43: C: Space missing after comma.
        time = time.change(:year=>now.year,:month=>now.month, :day=>now.day)
                                          ^
app/models/location.rb:199:5: C: Inconsistent indentation detected.
    def write_day_info(key, value) ...
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
script/scrub_data.rb:5:3: C: Put empty method definitions on a single line.
  def init_detailed_hours ...
  ^^^^^^^^^^^^^^^^^^^^^^^

186 files inspected, 7 offenses detected
```

Can you make sure all of these are addressed?

**Comment by tmikeschu** [comment]
@ericfreese : sorry about all this. I run rubocop before every push, so I'm not sure why stuff keeps popping up after the fact. Maybe the recent PR merges expose things differently.

Anyway, I've addressed those new offenses and it looks like Circle CI approves.

**Pull request merged**